### DISCUSSION TITLE
Filter measuring wrong results

### DISCUSSION THREAD

**Discussion by W0lpertinger** [jankae/LibreVNA#240] [General]
Hi,
a very simple thing to measure a lowpass filter between 500 kHz and 8 Mhz.

I calibrate SOLT on both ports, with the through cable I read S21 0,000dB as it should be.
The through cable is 30 cm with SMA, the connection to the DUT is 2x 15 cm one side each solderd to the PCB.
So exakt the same cable just cutted into two peaces.

With the DUT connected I can read 0.145 dB for S21 / S12.
This can not be right! It must be negative, an attenuation must be -0.145 dB.
Of course I get a negative reading in the stop band.

I tried 0 dBm and after a new calibration I got -0.428 dB S21 / S12.
This is also wrong, this is too high.

Measured with a RS FSH4 S21 is 0.08 dB, same Short Open Load and cable.

1st why is the resulut of S21 positive at -10 dBm?
2nd why is the result of S21 different at 0 dBM and too high?
3rd why is the result changing why changing the power for the measurement?
The calibration is done on each change of power.

The DUT is Chebyshev Lowpass Filter 5th order, nothing new.
Just want to do some measurings for a documentation.

Tried with 1.2.1 and 1.5.0. with the matching firmware for each version.

**Comment by jankae** [comment]
Hi,
this is difficult to judge without seeing the measurement setup and the calibration procedure. Here are some questions that could help identify issues:

- How is the through calibration standard defined? The 30 cm cable will have some loss and delay, is that accounted for in the standard definition?
- Did you actually cut the exact same cable you used for the calibration or is it an identical (other) cable?
- Are these results repeatable when you reboot everything and redo the calibration?

I wouldn't worry too much about a 0.145 dB deviation from the expected value. Yes, it shouldn't be there and yes, the R&S measurement gets you a better result but its price point is also vastly different from the LibreVNA.

The frequency range is also not ideal for the LibreVNA. It can take measurements there, but the dynamic range and accuracy will be better above 10 MHz.

The -0.428 dB is a bit more interesting. My first guess is that you are running into an overload condition here. Do you have an "ADC overload" warning in the bottom right corner? If this warning is there either during the calibration or during the DUT measurement, it could easily throw off your measurement by half a dB (or even more). You need to reduce the power to get rid of that warning.

One other thing I can think of is port mismatch on the LibreVNA which could result in some additional reflections on the cable. This will show up as valleys and peaks in the frequency response. Their locations change depending on the cable (or DUT) length. Inserting your DUT will change this and a peak may end up where a valley was before, resulting in a positive dB value. But in general, these reflections come into play above 3 GHz with the LibreVNA, so it is somewhat unlikely to be the problem here.

**Comment by W0lpertinger** [comment]
Hi,
the 30cm cables are from Rosenberger, no Chinese crap.
I cut an identical cable into two peaces of exact same length.
The results are repeatable.

I'm worrying a lot about the 0.145 dB in insertion loss!
It is called loss and I have a gain, thats the point the reading must be below 0 dB.
I'm not comparing RS with this VNA, of course not. But it should be clear that insertion loss
must be a loss and below 0 dB.
If somebody would show me measurements with an insertion loss witch is a "gain", means more than 0 dB I would
wonder a lot and think that this guy isn't working seriously.

I allways use -10 dBm, the 0 dBm was just a test where I've been wondering that every change in power with a new calibration is showing different results. This should not be!

Of course I reboot the computer several times and also the VNA after Firmware change from 1.2.1 - 1.5.0.
But there was noch change.


[SOLT_12_500k-8_00M_1000pt.zip](https://github.com/jankae/LibreVNA/files/14152189/SOLT_12_500k-8_00M_1000pt.zip)

**Comment by jankae** [comment]
> I'm worrying a lot about the 0.145 dB in insertion loss!
It is called loss and I have a gain, thats the point the reading must be below 0 dB.

Well, of course your filter will have no gain. What I meant is that I wouldn't worry about such a small deviation from the expected value (your filter has probably very little insertion loss). If you are really concerned with such precise measurements, the first thing I would do is create a valid calibration kit file. In your attached calibration, you are using ideal standard (the default in the GUI if you don't change that). This also means that your through is expected to be of length 0 and has no insertion loss at all.

This in itself should not cause a positive dB value for S21, it just introduces additional uncertainty because your standards likely are not ideal (although they are probably fairly close to ideal at such low frequencies).

One thing you could try to rule out or confirm my suspicion of port mismatch: attach attenuators (e.g. 3dB) at both ports of the VNA. Then repeat your calibration and measurement with these attenuators. If your measurements are now better, there could be some reflection from the VNA ports.

One other thing you could do to sanity check the measurement is measuring a different cable than your through standard (preferably approximately as long as your DUT) and check if that one will also have positive dB values.

**Reply by W0lpertinger** [reply]
Good Morning,
I tried different high quality cables before without any change in the results.
I have no available 3 dB attenuator, those I have are inside an SDR TRX, too much work to get them out...
For a first try and error I choose 10 dB attenuators.
Attached a measuerement with -10 dBm and 0 dBm, of course I did a new calibration on each change in power.
This looks both valid and acceptable!

How can I solve the port missmatch?

[Test_-10dB.zip](https://github.com/jankae/LibreVNA/files/14156260/Test_-10dB.zip)

**Reply by jankae** [reply]
Okay, looks like port mismatch is indeed the culprit here. There is not much you can do to improve the ports directly on the LibreVNA, that is a task for me when I manage to design a new hardware version.

What you can do is use the attenuators at the port. It is a trade-off between improving port reflections and sacrificing dynamic range. An X dB attenuator will decrease reflections at the port by 2X but it will also degrade SNR by 2X, limiting the noise floor of the VNA. 10 dB is probably a bit much, I would start with 3 dB or 6 dB and check if that results in good enough measurements for you. But if you don't care too much about dynamic range (or your DUTs stop band is still above the noise floor of the LibreVNA even with higher attenuation), 10 dB is also fine.

**Reply by W0lpertinger** [reply]
Today I tried again.
First with -10dBm and 3 dB att. Result is still positiv.
2nd try with 6 dB att. and now it is negative.
If it would be a mismatch, it should be possible to increase power and get the same negative result as before.
I set -3 dBm, new calbration and the 6 dB att as before.
Result is positive!
Hard to figure out what result is valid if changing the power change the result.

**Reply by jankae** [reply]
Can you post some numbers for these tests as well? It is hard to estimate the severity of this if I don't know whether it is 0.05 or 0.5 dB off from the expected value.
Also, what IF bandwidth are you using? Can you try if lowering that gives you better results? I know that you need a low bandwidth (or rather: a slow scan speed) for narrowband filter measurements but your filter doesn't feel like it is that slow. Still, might be worth a try.

I can just reiterate that I still wouldn't worry about these results (provided you don't have significantly higher positive values now than before). Yes, it seems weird to get positive dB values for a passive circuit but in the end it is all just the accuracy of the VNA. Would you worry about getting -2.85 dB reading from a filter that has -3 dB insertion loss? Getting +0.145 dB from a filter that has close to 0 dB insertion loss is the same thing.

**Reply by W0lpertinger** [reply]
As I wrote, to get positive values for insertion loss is not ok no matter how high the positive reading is.
I need some mearuements, the reason why I bought this VNA, to verify and show the results of my development.
Of course I'll not provide any measurement with positive insertion loss as I wrote above.
The measurements will be shown on the web and also in some papers.
As it is a kind of hobby I thought I do not need R&S for just measureing a filter.

Accuracy... insertion loss 0,3 dB or -0,2 dB, big deal!
63 dBm - 0, 5 dB = 62,5 dBm that means maybe 217W HF loss or not 217 W loss!
This loss means a lot of heat if it would be true!
As you see it is maybe no problem on small signals but with 63 dBm it is getting critical.

But coming back to the technical point.
The matching of your ports is maybe not good enough as you wrote.
With 6 dB att on each side it would be "better" while calibrating them out.
But I still dont understand why the result of insertion loss changes when the power setting changes.
Power setting does not effect the matching of your ports.

RBW is set to 1kHz / Power  -10 dBm

Please give me some instructions what I can measure to provide you usefull data to figure out whats going wrong.

**Reply by jankae** [reply]
> Please give me some instructions what I can measure to provide you usefull data to figure out whats going wrong.

For starters, the actual measurements with different attenuators at the ports and power levels instead of just saying that it is positive or negative.

> Accuracy... insertion loss 0,3 dB or -0,2 dB, big deal!
63 dBm - 0, 5 dB = 62,5 dBm that means maybe 217W HF loss or not 217 W loss!

I think it is also a stretch to extrapolate from -10 dBm to 63 dBm. I don't know how your filter is constructed and whether it is intended to be used at such a high power level. The components in your filter would need to be really, really linear for a measurement at -10 dBm to also be correct down to tenths of a dB at +63 dBm.

> As I wrote, to get positive values for insertion loss is not ok no matter how high the positive reading is.

I'd like to reword this slightly: it is impossible for a passive filter to have a positive gain. Whether you get a positive measurement on a VNA entirely depends on the accuracy of your calibration (which is in your hands) and also the base accuracy of the VNA (which you can not influence).
Let me give you an example with a completely different measurement: take a battery, a (very large) resistor and an ampere meter and connect everything in series. There will be a (very small) current flowing out of the battery. It is physically impossible for a current to flow into the battery since there is no other power source. Yet, the multimeter may show such a current if its accuracy is in the same order of magnitude as the current flowing out of the battery.

Every single VNA out there will have some uncertainty on its reported dB values. Just as an example, here is an excerpt from the specification of an SNA5014A (which costs about 25k):
![image](https://github.com/jankae/LibreVNA/assets/16818866/e1db5e6a-415f-440e-a5de-5fea7715d2df)
Your S12/S21 on that VNA may deviate from the actual value by 0.06 dB even after calibration.

I don't have a reliable value for the LibreVNA but I think it is safe to assume that it will be worse than that. In my (arguably limited) experience with RF measurements, you need to accept a few tenths of dB uncertainty or be prepared to spent significantly more on your test equipment.

> But I still dont understand why the result of insertion loss changes when the power setting changes.
Power setting does not effect the matching of your ports.

The power setting does not influence the matching but it shifts the operating point of the components in the VNA. Yes, the calibration should take care of that. But in reality, just by attaching your filter you will shift that operating point slightly. Combine that with the low frequency range (for which the LibreVNA is not optimized) and you will get the deviation you observe. I am reasonable sure that the measurement error would be smaller for e.g. 100-110 MHz.

I don't think I will engage in further discussions about whether a small positive dB value is acceptable. If you are worried about the scientific reputation of your results I'd suggest putting error bars on the values.

**Reply by W0lpertinger** [reply]
This filter is developed by myself 8 years ago for an amplifier.
I'm pretty sure I know what I'm doing.  With the right components it is not a big deal to handle this power and of course the values of the filter will not change between -10 dBm or 63 dBm! 

> Combine that with the low frequency range (for which the LibreVNA is not optimized) and you will get the deviation you observe. I am reasonable sure that the measurement error would be smaller for e.g. 100-110 MHz.

Thats ok if you would not give specification from 100 kHz - .... 
I didn't got the information that the VNA isn't working with lower performance on lower frewuencies.
What are lower frequencies, below 100 MHz? You should provide this information!

> The power setting does not influence the matching but it shifts the operating point of the components in the VNA. Yes, the calibration should take care of that. But in reality, just by attaching your filter you will shift that operating point slightly.

This is why a calibration is done!
If a SOLT calibration is done S12 / S21 can never be positiv with the DUT (passive) between the ports.

> I don't think I will engage in further discussions about whether a small positive dB value is acceptable. If you are worried about the scientific reputation of your results I'd suggest putting error bars on the values.

Ouch, that's maybe your way to putting error bars on the values.
This is not how serios people presenting measurements!
The point is NOT the accuracy it self with some tenth of dB, it is a useless mearuement of S12 / S21 with a positive result after calibrating.

As you wrote you don't want engage in further discussions about whether a small positive dB value is acceptable. 
Thats ok for me, peace!

Instead of blocking and fight for your opinion you should start to be openen minded and accapt user experience as an important input to find weak points on your product and solve them together with the user.

I'll write about my expierence in low frequency in our paper to share my impressions.
Not to blame you, you did a good job, so far.
But for S12 / S21 the measurements are not valid and nearby €700,- is too much for abitioned amateurs and hobbiest.

I'll go back for VNWA3SE I used before, al little slow but more valid results in lower frequencies.

**Reply by jankae** [reply]
> Instead of blocking and fight for your opinion you should start to be openen minded and accapt user experience as an important input to find weak points on your product and solve them together with the user.

I usually am, sorry that you feel I do not take you seriously here. But in this case my opinion is: you are running against the accuracy limit of the LibreVNA. That, together with your absolute unwillingness to accept the slightest positive dB value makes it hard for me to really help.

> Ouch, that's maybe your way to putting error bars on the values.
This is not how serios people presenting measurements!

I'd argue that all the most serious measurements will include error bars... No measurement is free from uncertainty and stating that uncertainty improves at least my trust in the result.

> Thats ok if you would not give specification from 100 kHz - ....
I didn't got the information that the VNA isn't working with lower performance on lower frewuencies.
What are lower frequencies, below 100 MHz? You should provide this information!

The dynamic range is actually spec'd worse for lower frequencies, see here:

![image](https://github.com/jankae/LibreVNA/assets/16818866/d47f0ec3-ae4e-4828-86ae-f917ac4b9ba4)

And with worse dynamic range also comes more noise (and uncertainty) at higher signal levels. Could the specifications for the LibreVNA be more extensive? Sure. But I do not have the equipment to fully characterize everything.

> I'll go back for VNWA3SE I used before, al little slow but more valid results in lower frequencies.

I think that is a good choice. I do not have personal experience with that VNA but have heard good things about it and just from the spec it is indeed better at lower frequencies than the LibreVNA.

**Comment by YuryW** [comment]
> How can I solve the port missmatch?


Hi,
The information in the following sections of 
the Handbook of Microwave Component Measurements: 
with Advanced VNA Techniques (Joel P. Dunsmore) might probably help:
5.1.3  Attenuation Measurements,
5.1.3.1  Connector Compensation Using Port Matching,
5.1.3.2  Connector Compensation Using Time Domain Filtering...

**Reply by W0lpertinger** [reply]
Thank you, I'll have a look next weekend wwith a little bit more time...
As this is a problem at the primary side (VNA Hardware) I must / should be solved there.
